aggregators/codec.go

```diff
@@ -63,7 +63,9 @@ func (k *CombinedMetricsKey) SizeBinary() int {
 // ToProto converts CombinedMetrics to its protobuf representation.
 func (m *CombinedMetrics) ToProto() *aggregationpb.CombinedMetrics {
 	pb := aggregationpb.CombinedMetricsFromVTPool()
-	pb.ServiceMetrics = make([]*aggregationpb.KeyedServiceMetrics, 0, len(m.Services))
+	if len(m.Services) > cap(pb.ServiceMetrics) {
+		pb.ServiceMetrics = make([]*aggregationpb.KeyedServiceMetrics, 0, len(m.Services))
+	}
```

Comment:
```suggestion
	pb.Services = growSlice(pb.Services, len(m.Services))
```

When we eventually update to Go 1.21, we can use https://pkg.go.dev/slices@master#Grow. For now we could copy the function locally:

```go
func growSlice[S ~[]E, E any](s S, n int) S {
	if n < 0 {
		panic("cannot be negative")
	}
	if n -= cap(s) - len(s); n > 0 {
		s = append(s[:cap(s)], make([]E, n)...)[:len(s)]
	}
	return s
}
```

WDYT?

Reply:
Sounds good to me. Any licensing implications to copy the function from go source code?